app/javascript/components/percentage-dropdown.vue

```diff
@@ -0,0 +1,65 @@
+<template>
+  <dropdown
+    ref="parentDropdown"
+    class="dropdown"
+  >
+    <div
+      class="dropdown__btn"
+      slot="btn"
+    >
+      <span class="dropdown__title">
+        {{ selectedItem ? selectedItem+'%' : noItemsMessage }}
+      </span>
+    </div>
+    <div
+      class="dropdown__body"
+      slot="body"
+    >
+      <div
+        v-if="bodyTitle"
+        class="dropdown-body__title"
+      >
+        {{ bodyTitle }}
+      </div>
+      <div
+        v-for="(item, index) in items"
+        class="dropdown__option"
+        @click="itemClicked({ item, index, user })"
+        :key="index"
+      >
+        {{ item+'%' }}
+      </div>
+    </div>
+  </dropdown>
+</template>
+
+<script>
+export default {
+  props: {
+    user: {
+      type: Object,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      bodyTitle: this.$i18n.t('message.froggoTeams.percentageDropdownTitle'),
+      items: ['20', '40', '50', '60', '80', '100'],
+      selectedPercentage: this.user.percentage,
+      noItemsMessage: '',
+    };
+  },
+  computed: {
+    selectedItem() {
+      return this.selectedPercentage;
+    },
+  },
+  methods: {
+    itemClicked(event) {
+      this.selectedPercentage = event.item;
+      this.$refs.parentDropdown.closeDropdown();
+      this.$emit('item-clicked', event);
```

Comment:
Me hace ruido que el evento se genere fuera del método y se pase entero como argumento. Sobre todo porque el user es una prop entonces lo puedes acceder con this.
me hace más sentido que el método sea `itemClicked(item)` y adentro armes el objeto, sobre todo si es que no estas usando el index en el componente donde recibes el evento.